File: Android_Projeto3_Gorjeta/app/src/main/java/com/atomiqui/gorjeta/MainActivity.java
package com.atomiqui.gorjeta;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.SeekBar;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {
    private TextInputEditText textInput_valor;
    private TextView text_porcent, text_exibeGorjeta, text_exibeTotal, text_aviso;
    private SeekBar seekBar_porcent;
    private double totalInicial = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textInput_valor     = findViewById(R.id.textInput_valor);
        text_porcent        = findViewById(R.id.text_porcent);
        text_exibeGorjeta   = findViewById(R.id.text_exibeGorjeta);
        text_exibeTotal     = findViewById(R.id.text_exibeTotal);
        text_aviso          = findViewById(R.id.text_aviso);
        seekBar_porcent     = findViewById(R.id.seekBar_porcent);

        seekBar_porcent.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int i, boolean b) {
                double gorjeta, porcentagem = seekBar.getProgress();

                try {
                    totalInicial = Double.parseDouble(textInput_valor.getText().toString());
                    gorjeta = (porcentagem/100)*totalInicial;

                    DecimalFormat df = new DecimalFormat("#.00");

                    text_porcent.setText( porcentagem + "%");
                    text_exibeGorjeta.setText("R$ " + df.format(gorjeta));
                    text_exibeTotal.setText("R$ " + df.format(totalInicial+gorjeta));

                    text_aviso.setText("");
                }
                catch (Exception e) {
                    seekBar_porcent.setProgress(0);
                    text_porcent.setText("0%");
                    text_exibeGorjeta.setText("");
                    text_exibeTotal.setText("");
                    text_aviso.setText("Insira um valor válido!");
                }
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
    }
}

File: Android_Projeto3.1_Gorjeta2/app/src/main/java/com/atomiqui/gorjetaemkotlin/MainActivity.kt
package com.atomiqui.gorjetaemkotlin

import android.os.Bundle
import android.widget.SeekBar
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputEditText
import java.text.DecimalFormat

class MainActivity : AppCompatActivity() {

    // Aparentemente não se faz isso, basta usar um cast que o tipo é atribuido automaticamente
    // private lateinit var textInput_preco: TextInputEditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // val pois essas variaveis não vai ser alteradas (nem devem)
        val textInput_preco = findViewById<TextInputEditText>(R.id.inputText_preco)
        val text_gorjeta = findViewById<TextView>(R.id.text_gorjeta)
        val text_total = findViewById<TextView>(R.id.text_total)
        val text_aviso = findViewById<TextView>(R.id.text_aviso)
        val text_percent = findViewById<TextView>(R.id.text_aviso)
        val seekBar = findViewById<SeekBar>(R.id.seekBar)

        seekBar?.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                val gorjeta: Double
                val porcentagem = seekBar!!.progress.toDouble()

                try {
                    var totalInicial : Double = textInput_preco.getText().toString().toDouble()
                    gorjeta = porcentagem / 100 * totalInicial
                    val df = DecimalFormat("#.00")

                    text_percent.setText("$porcentagem%")
                    text_gorjeta.setText("R$ " + df.format(gorjeta))
                    text_total.setText("R$ " + df.format(totalInicial + gorjeta))
                    text_aviso.text = ""
                }
                catch (e: Exception) {
                    seekBar.setProgress(0)
                    text_percent.setText("0%")
                    text_gorjeta.setText("")
                    text_total.setText("")
                    text_aviso.text = "Insira um valor válido!"
                }
                /*try {
                    var porcentagem : Int = seekBar!!.progress

                    var totalInicial : Double = textInput_preco.getText().toString().toDouble()
                    var gorjeta : Double = (porcentagem / 100) * totalInicial

                    val df = DecimalFormat("#.00")

                    text_percent.setText(porcentagem.toString() + "%")
                    text_gorjeta.setText("R$ " + df.format(gorjeta))
                    text_total.setText("R$ " + df.format(totalInicial + gorjeta))

                    text_aviso.text = ""
                }
                catch(e : Exception) {
                    seekBar?.setProgress(0)
                    text_percent.setText("0%")
                    text_gorjeta.setText("")
                    text_total.setText("")
                    text_aviso.text = "Insira um valor válido!"
                }*/
            }

            override fun onStartTrackingTouch(p0: SeekBar?) {
                //TODO("Not yet implemented")
            }

            override fun onStopTrackingTouch(p0: SeekBar?) {
                //TODO("Not yet implemented")
            }

        })

    }
}
